protoc_plugin/lib/src/shared.dart

```diff
@@ -2,8 +2,61 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:convert';
+
+import '../protoc.dart';
+
 const protobufImportPrefix = r'$pb';
 const asyncImportPrefix = r'$async';
 const coreImportPrefix = r'$core';
 const grpcImportPrefix = r'$grpc';
 const mixinImportPrefix = r'$mixin';
+
+extension FileDescriptorProtoExt on FileGenerator {
+  String? commentBlock(List<int> path) {
+    if (!options.generateComments) {
+      return null;
+    }
+
+    final bits = descriptor.sourceCodeInfo.location
+        .where((element) => element.path.toString() == path.toString())
+        .toList();
+
+    if (bits.length == 1) {
+      final match = bits.single;
+      if (match.hasLeadingComments()) {
+        return toDartComment(match.leadingComments);
+      }
+    }
+
+    if (bits.length > 1) {
+      throw 'WTH? ${descriptor.sourceCodeInfo.toProto3Json()}';
```

Comment:
I wonder if we should just print a warning (to stderr, stdout is used to print the generated code) and continue. `SourceCodeInfo` stuff does not affect runtime behavior of the protos so maybe a malformed `SourceCodeInfo` shouldn't block the operation.

We should also rephrase this with something helpful. Maybe something like "proto definition at $span has multiple source code infos".

Reply:
yup. fixed